spec/units-of-measure.md

```diff
@@ -1,138 +1,133 @@
 - Measures are supported by special syntax.
 
-The syntax of constants ([§4.3](basic-grammar-elements.md#constants)) is extended to support numeric constants with units of measure. The
-syntax of types is extended with measure type annotations.
+The syntax of constants ([§4.3](basic-grammar-elements.md#constants)) is extended to support numeric constants with units of measure. The syntax of types is extended with measure type annotations.
 
 ```fsgrammar
 measure-literal-atom :=
-    long-ident                  -- named measure e.g. kg
-    ( measure-literal-simp )    -- parenthesized measure, such as (N m)
+    long-ident                                  -- named measure e.g. kg
+    ( measure-literal-simp )                    -- parenthesized measure, such as (N m)
 
 measure-literal-power :=
     measure-literal-atom
-    measure-literal-atom ^ int32 -- power of measure, such as m^3
+    measure-literal-atom ^ int32                -- power of measure, such as m^3
 
 measure-literal-seq :=
     measure-literal-power
     measure-literal-power measure-literal-seq
 
 measure-literal-simp :=
-    measure-literal-seq         -- implicit product, such as m s^- 2
+    measure-literal-seq                         -- implicit product, such as m s^- 2
     measure-literal-simp * measure-literal-simp -- product, such as m * s^3
     measure-literal-simp / measure-literal-simp -- quotient, such as m/s^2
-    / measure-literal-simp      -- reciprocal, such as /s
-    1                           -- dimensionless
+    / measure-literal-simp                      -- reciprocal, such as /s
+    1                                           -- dimensionless
 
 measure-literal :=
-    _ -- anonymous measure
-    measure-literal-simp        -- simple measure, such as N m
+    _                                           -- anonymous measure
+    measure-literal-simp                        -- simple measure, such as N m
 
 const :=
     ...
-    sbyte < measure-literal >   -- 8 - bit integer constant
-    int16 < measure-literal >   -- 16 - bit integer constant
-    int32 < measure-literal >   -- 32 - bit integer constant
-    int64 < measure-literal >   -- 64 - bit integer constant
-    ieee32 < measure-literal >  -- single-precision float32 constant
-    ieee64 < measure-literal >  -- double-precision float constant
-    decimal < measure-literal > -- decimal constant
+    byte < measure-literal >                    -- 8 - bit unsigned integer constant
+    uint16 < measure-literal >                  -- 16 - bit unsigned integer constant
+    uint32 < measure-literal >                  -- 32 - bit unsigned integer constant
+    uint64 < measure-literal >                  -- 64 - bit unsigned integer constant
+    sbyte < measure-literal >                   -- 8 - bit integer constant
+    int16 < measure-literal >                   -- 16 - bit integer constant
+    int32 < measure-literal >                   -- 32 - bit integer constant
```

Comment:
We should use `int` and `uint` rather than `int32` and `uint32` (see the alias table near the end of this document).
(Or change the alias table.)

Reply:
There may be confusion here because of the terms used in the BNF-like grammar of the spec vs. actual types in the F# language (ie: `fsgrammar` vs `fsharp`, to borrow the Markdown fence-post terminology). I interpret, eg, the `int32` in `int32 < measure > ` as corresponding to `token int32 = ( int | xint ) 'l'` from [Lexical Analysis](lexical-analysis.md). If that's the correct interpretation, then `int` would correspond to `regexp int = ...`, which isn't really something that can be annotated with a measure. At least _think_ that's what the original doc was intending. That having been said: it's rather confusing... so maybe we can add explanatory text, or find some other way to make it more clear. I could also be completely wrong with my interpretation ¯\_(ツ)_/¯

Reply:
In the Lexical Analysis line, `int32` is the name of a lexical token type (a longer and perhaps less confusing name for it would be `int32Literal`). While here, `int32` is a literal name of a F# type. Its usage in the `fsgrammar` block is just a verbatim string that the parser matches (see §1.2 for this notation).

So, after having written this, I think we would have to include both `int32` and `int` in the grammar (and also the other allowed type abbreviations). (Note that other versions of naming that type, like `System.Int32` are not allowed.)

Reply:
So, would this mean `ieee32` and `ieee64` (see lines: 47-48) should be replaced with `float32` and `float64`, respectively?

Reply:
Still thinking about this. I am not sure if what I wrote above is correct.

Reply:
No worries. 🙂 
Though the question still remains: do we try to somehow prevent others from going through the same head-scratching we just experienced?

Reply:
Where I was stuck, I believe, is that the grammar box defines measures, and inbetween `const`, but no other use of measures. Perhaps we should first define measures (but not yet const), then measure type definition and measure application
```
measure-type-definition := type measure = measure
primitive-numeric-type := byte | sbyte | ....
measure-application := primitive-numeric-type < measure >
```
(not sure if the above is completely correct)
and then `const`.

Reply:
But the above will then also need adding the items to the type-def and type grammars, and I don't know how deep the rabbit hole will be.
And since our main goal is not a generally improved spec, but the addition of the RFCs, I guess we should for now leave it as is.